Keda.Cosmosdb.Scaler/src/Services/CosmosDBExternalScaler.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Threading.Tasks;
+using Google.Protobuf.Collections;
+using Grpc.Core;
+using Keda.Cosmosdb.Scaler;
+using Keda.CosmosDB.Scaler.Protos;
+using Microsoft.Extensions.Logging;
+using static Keda.CosmosDB.Scaler.Protos.ExternalScaler;
+using Keda.CosmosDB.Scaler.Extensions;
+using Keda.CosmosDB.Scaler.Repository;
+using Keda.Cosmosdb.Scaler.Repository;
+
+namespace Keda.CosmosDB.Scaler.Services
+{
+    public class CosmosDBExternalScaler : ExternalScalerBase
+    {
+        private readonly ILogger _logger;
+        private readonly ICosmosDBRepository _cosmosDBRepository;
+
+        public CosmosDBExternalScaler(ILoggerFactory loggerFactory, ICosmosDBRepository cosmosDBRepository)
+        {
+            _logger = loggerFactory.CreateLogger<CosmosDBExternalScaler>();
+            _cosmosDBRepository = cosmosDBRepository;
+        }
+
+        public override async Task<IsActiveResponse> IsActive(ScaledObjectRef request, ServerCallContext context)
+        {
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScalerMetadata);
+                var workToBeDone = await GetEstimatedWork(trigger);
+
+                bool isActive = workToBeDone > 0;
+
+                if (isActive)
+                {
+                    _logger.LogDebug("Activating to 1 instance for Azure Cosmos DB account {accountName} with database {databaseName} and collection {collectionName}",
+                        trigger.AccountName, trigger.DatabaseName, trigger.CollectionName);
+                }
+
+                return new IsActiveResponse
+                {
+                    Result = isActive
+                };
+            }
+            catch (Exception)
+            {
+                throw;

```

Comment:
if the purpose it just throw - remove try ..catch.. unhandled exceptions will be bubbled